noir/noir-repo/acvm-repo/acvm/src/pwg/mod.rs

```diff
@@ -462,12 +544,37 @@ impl<'a, B: BlackBoxFunctionSolver> ACVM<'a, B> {
             }
             BrilligSolverStatus::Finished => {
                 // Write execution outputs
-                solver.finalize(witness, outputs)?;
+                solver.finalize(&mut self.witness_map, outputs)?;
                 Ok(None)
             }
         }
     }
 
+    fn map_brillig_error(&self, mut err: OpcodeResolutionError) -> OpcodeResolutionError {
```

Comment:
It doesn't seem like we make use of `&self` in here as far as I can see?

Reply:
It uses `self.assertion_payloads`

Reply:
Ah, so it does. Too used to `self` being highlit in red from Noir.